pages/merchant/_reducer.tsx

```diff
@@ -0,0 +1,103 @@
+import router from "next/router"
+import React from "react"
+
+export const ACTIONS = {
+  ADD_DIGIT: "add-digit",
+  DELETE_DIGIT: "delete-digit",
+  CLEAR_INPUT: "clear-input",
+  CREATE_INVOICE: "create-invoice",
+  CREATE_NEW_INVOICE: "create-new-invoice",
+  GET_PAYMENT_STATUS: "get-payment-status",
+  BACK: "back-by-one-history",
+}
+
+export type ACTIONTYPE = { type: string; payload?: string | (() => void) }
+
+function reducer(state: React.ComponentState, { type, payload }: ACTIONTYPE) {
+  switch (type) {
+    case ACTIONS.ADD_DIGIT:
+      if (payload == "0" && state.currentAmount == "0") return state
+      if (payload === "." && state.currentAmount.includes(".")) return state
+      if (state.currentAmount?.length >= 14) return state
+      return {
+        ...state,
+        currentAmount: `${state.currentAmount || ""}${payload}`,
+      }
+
+    case ACTIONS.DELETE_DIGIT:
+      if (state.currentAmount == null) return state
+      return {
+        ...state,
+        currentAmount: state.currentAmount?.slice(0, -1),
+      }
+
+    case ACTIONS.CLEAR_INPUT:
+      if (state.currentAmount == null) return
+      if (state.username == null) return
+      router.push(
```

Comment:
Shouldn't reducers avoid side effects?

Reply:
I've resolved this by just returning the state